docs/limitations.rst

```diff
@@ -14,6 +17,38 @@ certain limitations:
   class like ``Eigen::Array`` or ``boost.multi_array``.
 
 These features could be implemented but would lead to a significant increase in
-complexity. I've decided to draw the line here to keep this project simple and
-compact. Users who absolutely require these features are encouraged to fork
-pybind11.
+complexity. Pybind11 strives to be simple and compact. Users who absolutely
+require these features are encouraged to fork pybind11.
```

Comment:
Reading this hurts a little. Instead, I'd encourage them to suggest plug-in-kind-of mechanisms (like the "If you know how" just below), so they can layer their functionality on top. This must be possible somehow, without forking, in the worst case with `#ifdef`s, no?

Reply:
I've reworded this quite a bit now, I originally left it (almost) alone. Includes a pointer to an example now too.

Reply:
That's awesome! — There is another one like this in the pipeline, much closer to home, currently https://github.com/pybind/pybind11_abseil/tree/experimental. Just mentioning FYI, I don't think we should mention it here at the moment.

Reply:
I noticed that, and figured you'd know something about it. ;)